code/modules/mob/dead/observer/observer.dm

```diff
@@ -107,6 +107,8 @@
 
 	if(!name) //To prevent nameless ghosts
 		name = capitalize(pick(first_names_male)) + " " + capitalize(pick(last_names))
+	if(name == "Unknown")
+		name = body.real_name
```

Comment:
recipe for trolling -> set name to "unknown" -> observe -> you have no body and this forces a runtime

add sanity check please

Reply:
you can't even name yourself Unknown for some reason, but I've done as you asked